projects/extension/src/background/ConnectionManager.ts

```diff
@@ -175,8 +196,18 @@ export class ConnectionManager extends (EventEmitter as {
       .filter((c) => c.tabId === tabId && !c.parachain && c.chain)
       .map((c) => c.chain!)
 
+    const chainName = knownChainName?.toLowerCase()
+    const databaseContent =
+      chainName &&
+      (await new Promise<string>((res) =>
+        chrome.storage.local.get([chainName], (val) => {
+          return res(val[chainName])
+        }),
+      ))
```

Comment:
I don't  think that this logic belongs here, either. `addChain` should have an optional argument `databaseContent` and this content should only be retrieved either when there is an `add-well-known-chain` message or from the `init` script for the initial chains.